hs_core/tests/api/rest/test_folders.py

```diff
@@ -0,0 +1,113 @@
+import json
+from rest_framework import status
+from hs_core.models import ResourceFile
+# from django_irods.storage import IrodsStorage
+from hs_core.models import get_path
+
+
+from .base import HSRESTTestCase
+
+
+class TestFolders(HSRESTTestCase):
+
+    def test_create_folder(self):
+        rtype = 'GenericResource'
+        title = 'My Test resource'
+        params = {'resource_type': rtype,
+                  'title': title,
+                  'file': ('cea.tif',
+                           open('hs_core/tests/data/cea.tif'),
+                           'image/tiff')}
+        url = '/hsapi/resource/'
+        response = self.client.post(url, params)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        content = json.loads(response.content)
+        res_id = content['resource_id']
+        self.resources_to_delete.append(res_id)
+
+        url2 = str.format('/hsapi/resource/{}/folders/foo/', res_id)
+
+        # should not be able to ls non-existent folder
+        response = self.client.get(url2, {})
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        content = json.loads(response.content)
+        self.assertEqual(content, 'Cannot list path')
+
+        # should not be able to delete non-existent folder
+        response = self.client.delete(url2, {})
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        content = json.loads(response.content)
+        self.assertEqual(content, "Cannot remove folder")
+
+        # create a folder
+        response = self.client.put(url2, {})
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # should be able to create it again
+        response = self.client.put(url2, {})
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # list that folder: should work, should be empty
+        response = self.client.get(url2, {})
+        content = json.loads(response.content)
+        self.assertEqual(len(content['folders']), 0)
+        self.assertEqual(len(content['files']), 0)
+
+        # delete that folder: should be possible
+        response = self.client.delete(url2, {})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
```

Comment:
Would be good to add a ``get`` call to the deleted folder here and test the response.